File: hrms-backend/src/main/java/com/hrms/service/PerformanceService.java

//service/PerformanceService.java
package com.hrms.service;

import com.hrms.model.EmployeeKra;
import com.hrms.model.PerformanceCycle;
import java.util.List;
import java.util.Map;

public interface PerformanceService {
 PerformanceCycle createCycle(PerformanceCycle cycle);
 List<PerformanceCycle> getAllCycles();
 PerformanceCycle updateCycleStatus(Long id, String status);

 EmployeeKra createKra(EmployeeKra kra);
 List<EmployeeKra> getEmployeeKras(Long employeeId, Long cycleId);
 EmployeeKra updateKra(Long id, EmployeeKra kra);
 EmployeeKra submitSelfReview(Long id, Integer rating, String comments);
 EmployeeKra submitManagerReview(Long id, Integer rating, String comments);

 Map<String, Object> getEmployeePerformanceSummary(Long employeeId, Long cycleId);
}

File: hrms-frontend/src/services/performanceService.js
// src/services/performanceService.js
import api from '../api/axiosConfig';

const performanceService = {
  // Cycles
  createCycle: (data) => api.post('/performance/cycles', data),
  getAllCycles: () => api.get('/performance/cycles'),
  updateCycleStatus: (id, status) =>
    api.patch(`/performance/cycles/${id}/status?status=${status}`),

  // KRAs
  createKra: (data) => api.post('/performance/kras', data),
  getKras: (empId, cycleId) =>
    api.get(`/performance/kras/${empId}/${cycleId}`),
  updateKra: (id, data) => api.put(`/performance/kras/${id}`, data),
  submitSelfReview: (id, data) =>
    api.post(`/performance/kras/${id}/self-review`, data),
  submitManagerReview: (id, data) =>
    api.post(`/performance/kras/${id}/manager-review`, data),

  // Summary
  getSummary: (empId, cycleId) =>
    api.get(`/performance/summary/${empId}/${cycleId}`),
};

export default performanceService;